core/drivers/clk/clk.c

```diff
@@ -70,7 +70,10 @@ struct clk *clk_lw_alloc(const struct clk_ops *ops, size_t count)
 
 void clk_free(struct clk *clk)
 {
-	free(clk);
+	if (clk && is_clk_elt(clk))
```

Comment:
During this review I've seen a reluctance to add operators. Instead has the first approach often been like here, that is, test with `is_clk_elt()` and do special treatment. With operators you're able to take a more object oriented approach where the object (struct clk) provides operators for the needed special treatment. Is this reluctance because it's unexpected that we'll extend with other types of `struct clk_xxx` or because a hybrid approach seems easier right now?
 
Wouldn't it be easier with an operator for this?
``` C
if (clk)
    clk->ops->free(clk);
```

If you'd rather save an operator please add a comment somewhere so users of this framework understands how to use it.

Reply:
makes sense.
i face similar concern when considering other kind of drivers.
i'll add a `release` operator (`free` to too malloc() scope to me here).

Reply:
Well, actually `free` is fine, API function is already `clk_free()`...

Reply:
see fixup commit https://github.com/OP-TEE/optee_os/pull/4978/commits/3ce0d85e5733243ad8ede5a7bfa84e3d50417789.